apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -148,21 +181,37 @@ private string ValidatedDataSource(string dataSource)
 
         /// <summary>
         /// The TCP Host name to connect to Spanner. If not supplied in the connection string, the default
-        /// host will be used.
+        /// host will be used. If the SPANNER_EMULATOR_HOST environment variable is set, the host specified
+        /// in its value will be used as the default host.
         /// </summary>
         public string Host
         {
-            get => GetValueOrDefault(nameof(Host), SpannerClient.DefaultEndpoint.Host);
+            get
+            {
+                if (ContainsKey(nameof(Host)))
+                {
+                    return (string)this[nameof(Host)];
+                }
+
+                ParseEmulatorHostAndPort();
+                return _emulatorHost ?? SpannerClient.DefaultEndpoint.Host;
+            }
             set => this[nameof(Host)] = value;
         }
 
         /// <summary>
         /// The TCP port number to connect to Spanner. If not supplied in the connection string, the default
-        /// port will be used.
+        /// port will be used.  If the SPANNER_EMULATOR_HOST environment variable is set, the port specified
+        /// in the value will be used as the default port.
         /// </summary>
         public int Port
         {
-            get => GetInt32OrDefault(nameof(Port), 1, 65535, SpannerClient.DefaultEndpoint.Port);
+            get
+            {
+                ParseEmulatorHostAndPort();
+                return GetInt32OrDefault(nameof(Port), 1, 65535,
+                    (string.IsNullOrEmpty(_emulatorHost)) ? SpannerClient.DefaultEndpoint.Port : _emulatorPort);
```

Comment:
nit: You can check for null only, and if you change the type of `_emulatorPort` to `int?` you can do `_emulatorPort ?? Spanner.DefaultEndpoint.Port`

Reply:
#TIL